crates/rpc/rpc-api/src/debug.rs

```diff
@@ -359,6 +360,15 @@ pub trait DebugApi {
     #[method(name = "startGoTrace")]
     async fn debug_start_go_trace(&self, file: String) -> RpcResult<()>;
 
+    /// Returns the state root of the `HashedPostState` on top of the state for the given block with
+    /// trie updates.
+    #[method(name = "stateRootWithUpdates")]
+    async fn debug_state_root_with_updates(
+        &self,
+        hashed_state: HashedPostState,
+        block_id: Option<BlockId>,
+    ) -> RpcResult<(B256, TrieUpdates)>;
```

Comment:
I believe this will serializes as [0x, {}] which is perhaps a bit weird, should this be a single object instead, couldn't find this in the referenced pr